conjure-java-jaxrs-client/src/main/java/com/palantir/conjure/java/client/jaxrs/feignimpl/MethodHeaderEnrichmentContract.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * (c) Copyright 2017 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.client.jaxrs.feignimpl;
+
+import feign.Contract;
+import feign.MethodMetadata;
+import java.lang.reflect.Method;
+
+/**
+ * Contract to capture the method name and pass it to a feign client.
+ *
+ * This should be considered internal API and should not be depended upon.
+ */
+public final class MethodHeaderEnrichmentContract extends AbstractDelegatingContract {
```

Comment:
`AbstractDelegatingContract` is currently package-private, so we have to pick our poison. Would we rather make this public of make `AbstractDelegatingContract` public?

Reply:
Ah, fair enough — I think it’s good as is

Reply:
I took at stab at this in https://github.com/palantir/conjure-java-runtime/pull/2973/commits/80e54cae7c8b17d472fb4f2910042097fa979127.

It will allow us to make specific contracts/encoders/decoders package-private at the expense of adding a couple public utility methods. Let me know which you'd prefer. I'm pretty indifferent, maybe with a slight preference for leaving this how it was.

Reply:
Let’s keep it simple — agree that the minimal change is probably ideal